File: test-video.ts
import { VideoGenerator } from "./src/video-generator";
import { PromptGenerator } from "./src/prompt-generator";

async function testVideoGeneration() {
  console.log("🧪 Testing video generation...");
  
  const provider = (process.env.VIDEO_PROVIDER as any) || "veo2";
  console.log(`Using provider: ${provider}`);
  
  const videoGen = new VideoGenerator(
    process.env.GEMINI_API_KEY!,
    process.env.REPLICATE_API_KEY!,
    process.env.VERTEX_API_KEY!,
    process.env.GOOGLE_PROJECT_ID!,
    provider
  );
  
  const promptGen = new PromptGenerator(process.env.OPENAI_API_KEY!);
  
  try {
    console.log("Generating a test prompt...");
    const prompt = await promptGen.generatePrompt();
    console.log(`Generated prompt: "${prompt.text}"`);
    
    console.log("Generating video from prompt...");
    const video = await videoGen.generateVideo(prompt);
    
    console.log("✅ Video generation successful!");
    console.log(`Video ID: ${video.id}`);
    console.log(`Video URL: ${video.videoUrl}`);
    console.log(`Duration: ${video.duration} seconds`);
    
    return video;
  } catch (error) {
    console.error("❌ Video generation failed:", error);
    throw error;
  }
}

async function testCustomPrompt() {
  console.log("\n🎨 Testing custom prompt...");
  
  const videoGen = new VideoGenerator(
    process.env.GEMINI_API_KEY!,
    process.env.REPLICATE_API_KEY!,
    process.env.VERTEX_API_KEY!,
    process.env.GOOGLE_PROJECT_ID!,
    (process.env.VIDEO_PROVIDER as any) || "veo2"
  );
  
  const customPrompt = "A sophisticated orange tabby cat wearing tiny glasses, sitting at a miniature desk, looking directly at the camera with an expression that says 'I've seen things you wouldn't believe.' The cat occasionally adjusts its glasses and sighs dramatically.";
  
  try {
    console.log(`Testing custom prompt: "${customPrompt}"`);
    const video = await videoGen.testGeneration(customPrompt);
    
    console.log("✅ Custom prompt test successful!");
    console.log(`Video URL: ${video.videoUrl}`);
    
    return video;
  } catch (error) {
    console.error("❌ Custom prompt test failed:", error);
    throw error;
  }
}

if (import.meta.main) {
  Promise.all([
    testVideoGeneration(),
    testCustomPrompt()
  ]).then(() => {
    console.log("\n🎉 All tests completed!");
  }).catch((error) => {
    console.error("\n💥 Tests failed:", error);
    process.exit(1);
  });
}

File: test_video.go
package main

import (
	"context"
	"fmt"
	"log"
	"os"
)

func testVideoGeneration() {
	fmt.Println("🧪 Testing video generation...")

	ctx := context.Background()

	provider := VideoProvider(getEnvWithDefault("VIDEO_PROVIDER", "veo2"))
	fmt.Printf("Using provider: %s\n", provider)

	videoGen, err := NewVideoGenerator(
		os.Getenv("GEMINI_API_KEY"),
		os.Getenv("REPLICATE_API_KEY"),
		os.Getenv("VERTEX_API_KEY"),
		os.Getenv("GOOGLE_PROJECT_ID"),
		provider,
	)
	if err != nil {
		log.Fatalf("Failed to create video generator: %v", err)
	}

	promptGen := NewPromptGenerator(os.Getenv("OPENAI_API_KEY"))

	// Test 1: Generate prompt and video
	fmt.Println("Generating a test prompt...")
	prompt, err := promptGen.GeneratePrompt(ctx)
	if err != nil {
		log.Fatalf("Failed to generate prompt: %v", err)
	}

	fmt.Printf("Generated prompt: \"%s\"\n", prompt.Text)

	fmt.Println("Generating video from prompt...")
	video, err := videoGen.GenerateVideo(ctx, prompt)
	if err != nil {
		log.Fatalf("Failed to generate video: %v", err)
	}

	fmt.Println("✅ Video generation successful!")
	fmt.Printf("Video ID: %s\n", video.ID)
	fmt.Printf("Video URL: %s\n", video.VideoURL)
	fmt.Printf("Duration: %d seconds\n", video.Duration)

	// Test 2: Custom prompt
	fmt.Println("\n🎨 Testing custom prompt...")
	customPrompt := "A sophisticated orange tabby cat wearing tiny glasses, sitting at a miniature desk, looking directly at the camera with an expression that says 'I've seen things you wouldn't believe.' The cat occasionally adjusts its glasses and sighs dramatically."

	fmt.Printf("Testing custom prompt: \"%s\"\n", customPrompt)
	customVideo, err := videoGen.TestGeneration(ctx, customPrompt)
	if err != nil {
		log.Fatalf("Failed to generate custom video: %v", err)
	}

	fmt.Println("✅ Custom prompt test successful!")
	fmt.Printf("Video URL: %s\n", customVideo.VideoURL)

	fmt.Println("\n🎉 All tests completed!")
}

func init() {
	// This allows the file to be run as a standalone test
	if len(os.Args) > 1 && os.Args[1] == "test-video" {
		testVideoGeneration()
		os.Exit(0)
	}
}